File: deno/routes/streamTranslation.js
import { Router } from "@oak/oak";

import { makeRequestToYandex } from "../requests.js";
import { validateJSONRequest } from "../validators.js";

const streamTranslationRouter = new Router()
  .post("/translate-stream", async (ctx) => {
    const [body, headers] = await validateJSONRequest(ctx);
    if (!body || !headers) {
      return;
    }

    // if the value is different from key:value, it will throw 500 error
    return await makeRequestToYandex(
      ctx,
      "stream-translation/translate-stream",
      body,
      headers
    );
  })
  .post("/ping-stream", async (ctx) => {
    const [body, headers] = await validateJSONRequest(ctx);
    if (!body || !headers) {
      return;
    }

    // if the value is different from key:value, it will throw 500 error
    return await makeRequestToYandex(
      ctx,
      "stream-translation/ping-stream",
      body,
      headers
    );
  });

export default streamTranslationRouter;

File: axum/src/routes/stream_translation.rs
use axum::{http::HeaderMap, response::IntoResponse, routing::post, Router};
use reqwest::Method;

use crate::utils::handlers::request_browser_bytes;

async fn post_strans_translate_stream(headers: HeaderMap, body: String) -> impl IntoResponse {
    request_browser_bytes(
        "/stream-translation/translate-stream",
        headers,
        body,
        Method::POST,
    )
    .await
}

async fn post_strans_ping_stream(headers: HeaderMap, body: String) -> impl IntoResponse {
    request_browser_bytes(
        "/stream-translation/ping-stream",
        headers,
        body,
        Method::POST,
    )
    .await
}

pub fn get_router() -> Router {
    Router::new()
        .route("/translate-stream", post(post_strans_translate_stream))
        .route("/ping-stream", post(post_strans_ping_stream))
}
